javaslang/src/main/java/javaslang/collection/Tree.java

```diff
@@ -747,13 +755,18 @@ public boolean isEmpty() {
         }
 
         @Override
+        public int length() {
+            return size;
+        }
+
+        @Override
         public boolean isLeaf() {
-            return children.isEmpty();
+            return size == 1;
         }
 
         @Override
-        public int size() {
-            return 1 + children.foldLeft(0, (acc, child) -> acc + child.length());
+        public int nodeCount() {
+            return size;
         }
```

Comment:
just one commit away from merging the PR :-) we do not need this method here because Tree has a default nodeCount() method.